### PR TITLE
Classify and extend convolution and depthwise performance tests

### PR DESCRIPTION
**Pull request by WanliZhong**
This PR aims to:
1. Extend the test cases from models: `YOLOv5`, `YOLOv8`, `EfficientNet`, `YOLOX`, `YuNet`, `SFace`, `MPPalm`, `MPHand`, `MPPose`, `ViTTrack`, `PPOCRv3`, `CRNN`, `PPHumanSeg`. (371 new test cases are added)

2. Classify the existing convolution performance test to below cases
    - CONV_1x1
    - CONV_3x3_S1_D1 (winograd)
    - CONV
    - DEPTHWISE

3. Reduce unnecessary test cases by follow 3 rules (366 test cases are pruned):
(i). For all tests, except for pad and bias related parameters, all other parameters are the same. Only one case can be reserved.
(ii). When the only difference is the channel of input shape, and other parameters are the same. Only one case can be reserved in each range `[1, 3], [4, 7], [8, 15], [16, 31], [32, 63], [64, 127], [128, 255], [256, 511], [512, 1023], [1024, 2047], [2048, 4095]`
(iii). When the only difference is the width and height of input shape, and other parameters are the same. Only one case can be reserved in each range `[1, 31], [32, 63], [64, 95]... `

> **Reproduced**: 1. follow step in https://github.com/alalek/opencv/commit/dnn_dump_conv_kernels to dump all convolution cases from new models. (declared flops may not right, need to be checked manually) 2 and 3. Use the script from python code [classify conv.txt](https://github.com/opencv/opencv/files/13522228/classify.conv.txt)

**Performance test result on Apple M2**

**Test result details**:  [M2.md](https://github.com/opencv/opencv/files/13379189/M2.md)

**Additional test result details with FP16**:  [m2_results_with_fp16.zip](https://github.com/opencv/opencv/files/13491070/m2_results_with_fp16.zip)

**Brief summary for 4.8.1 vs 4.7.0 or 4.6.0**: 
1. `CONV_1x1_S1_D1` dropped significant with small or large input shape.
2. `DEPTHWISE_5x5 ` dropped a little compared with 4.7.0. 

---

**Performance test result on [Intel Core i7-12700K](https://www.intel.com/content/www/us/en/products/sku/134594/intel-core-i712700k-processor-25m-cache-up-to-5-00-ghz/specifications.html)**: 8 Performance-cores (3.60 GHz, turbo up to 4.90 GHz), 4 Efficient-cores (2.70 GHz, turbo up to 3.80 GHz), 20 threads.

**Test result details**: [INTEL.md](https://github.com/opencv/opencv/files/13374093/INTEL.md)
**Brief summary for 4.8.1 vs 4.5.5**: 
1. `CONV_5x5_S1_D1` dropped significant. 
2. `CONV_1x1_S1_D1`, `CONV_3x3_S1_D1`, `DEPTHWISE_3x3_S1_D1`, `DEPTHWISW_3x3_S2_D1` dropped with small input shape.

---

TODO:
- [x] Perform tests on arm with each opencv version
- [x] Perform tests on x86 with each opencv version
- [x] Split each test classification with single test config
- [x] test enable fp16

### CODE REVIEW

**Review by fengyuentau — COMMENTED** [review]
Could you give a summary on what have been removed and added?

**Comment by WanliZhong** [comment]
@fengyuentau Totally add 292 new cases, remove 287 old cases. You can check each cases in file [diff.txt](https://github.com/opencv/opencv/files/13371912/diff.txt)

**Comment by WanliZhong** [comment]
**UPDATE:** Test results are attached. I have finished the convolution performance test with each OpenCV release version. The result show depthwise may not the biggest problem. Many performance issues have been fixed at 4.8.1

**Comment by zihaomu** [comment]
Great job! Looks like we are no good on the small output shape and small input or output channle. With these performance tests, we can control compute branches in more detail.

**Comment by vpisarev** [comment]
@WanliZhong, we need to split this test into several cases: 1x1 convolution, 3x3s1d1 (winograd and im2row-based), depthwise, generic (the remaining cases). each convolution case should test FP32 and FP16

**Review by asmorkalov — APPROVED** [review]
:+1:

File: modules/dnn/perf/perf_convolution.cpp [hunk 1]
```diff
@@ -878,17 +882,106 @@ PERF_TEST_P_(Conv, conv)
...
+typedef TestBaseWithParam<ConvTestParam_t> Conv_Depthwise;
+
+PERF_TEST_P_(Conv, conv)
+{
+    const ConvParam_t& params = get<0>(GetParam());
+    Backend backendId = get<0>(get<2>(GetParam()));
+    Target targetId = get<1>(get<2>(GetParam()));
+    bool winograd = get<1>(GetParam());
+    Net net = build_net(params, backendId, targetId);
+    net.enableWinograd(winograd);
```
Comment (opencv-alalek) [inline modules/dnn/perf/perf_convolution.cpp:903]: There is "warmup" stage in the original test code.

If we change settings, then we should do that again.

Comment (WanliZhong) [reply modules/dnn/perf/perf_convolution.cpp:903]: warm up happen on `build_net()` function. I didn't change this part of the code

**Comment by vpisarev** [comment]
@WanliZhong, Winograd is only valid for 3x3s1d1; for other tests it does not make any sense. Could you please adjust your tests, otherwise we will have many useless cases for 1x1, depthwise, generic etc.

File: modules/dnn/perf/perf_convolution.cpp [hunk 2]
```diff
@@ -878,17 +882,106 @@ PERF_TEST_P_(Conv, conv)
         << "    Weights(parameters): " << divUp(weightsMemory, 1u<<10) << " Kb"
         << "    MFLOPS=" << flops * 1e-6 << std::endl;
 
+    EXPECT_NEAR(flops, declared_flops, declared_flops * 1e-6);
+
+    return net;
+}
+
+typedef tuple<ConvParam_t, bool, tuple<Backend, Target> > ConvTestParam_t;
```
Comment (vpisarev) [inline modules/dnn/perf/perf_convolution.cpp:890]: do
```
typedef tuple<ConvParam_t, tuple<Backend, Target> > ConvTestParam_t;
typedef tuple<ConvParam_t, tuple<Backend, Target>, bool> Conv3x3S1D1TestParam_t;
typedef TestBaseWithParam<ConvTestParam_t> Conv;
typedef TestBaseWithParam<ConvTestParam_t> Conv_1x1;
typedef TestBaseWithParam<Conv3x3S1D1TestParam_t> Conv_3x3S1D1;
typedef TestBaseWithParam<ConvTestParam_t> Conv_Depthwise;
```

File: modules/dnn/perf/perf_convolution.cpp (hunk 1 above)
Comment (opencv-alalek) [reply modules/dnn/perf/perf_convolution.cpp]: If you play with network configuration setting (like `net.enableWinograd(winograd);` on the line 932) then you should do "warmup" again.

Comment (WanliZhong) [reply modules/dnn/perf/perf_convolution.cpp]: Thanks! It's right. I will do it soon

**Pull request merged**